src/ubx.cpp

```diff
@@ -1664,6 +1666,23 @@ GPSDriverUBX::payloadRxDone()
 		ret = 1;
 		break;
 
+	case UBX_MSG_TIM_TIM2:
+		UBX_TRACE_RXMSG("Rx TIM-TIM2");
+
+		{
+			// convert to unix timestamp
+			uint64_t tv_sec = _buf.payload_rx_tim_tim2.wnR * 7 * 24 * 3600;
+
+			_event_info->time_utc_usec = static_cast<uint64_t>(tv_sec) * 1e6 +
+						static_cast<uint64_t>_buf.payload_rx_tim_tim2.towMsR * 1e3 +
+						_buf.payload_rx_tim_tim2.towSubMsF / 1e3;
+			_event_info->timestamp = gps_absolute_time();
+
+		}
+
+		ret = 1;
```

Comment:
still return 1?

Reply:
We'll need to add another bit, so return 4.

Reply:
Thanks. It have been fixed.